src/main/java/net/mcreator/generator/blockly/BlocklyBlockCodeGenerator.java

```diff
@@ -220,6 +222,128 @@ public void generateBlock(BlocklyToCode master, Element block) throws TemplateGe
 			}
 		}
 
+		// next we check for input groups if they are defined, we process them and add to data model
+		if (!toolboxBlock.getRepeatingInputs().isEmpty()) {
+			for (String inputName : toolboxBlock.getRepeatingInputs()) {
+				Map<String, Element> matchingElements = elements.stream()
+						.filter(e -> e.getNodeName().equals("value") && e.getAttribute("name").startsWith(inputName)
+								&& e.getAttribute("name").substring(inputName.length()).replaceAll("\\d", "")
+								.equals("")).collect(Collectors.toMap(e -> e.getAttribute("name"), e -> e));
+				Map<Integer, String> processedElements = new HashMap<>();
+				int idx = 0;
+				while (!matchingElements.isEmpty()) {
+					if (matchingElements.containsKey(inputName + idx)) {
+						String generatedCode = BlocklyToCode.directProcessOutputBlock(master,
+								matchingElements.remove(inputName + idx));
+						processedElements.put(idx, generatedCode);
+					} else {
+						processedElements.put(idx, null); // we add null at this index to not shift other elements
+						master.addCompileNote(new BlocklyCompileNote(BlocklyCompileNote.Type.ERROR,
+								L10N.t("blockly.errors.input_empty", inputName + idx, type)));
+					}
+					idx++;
+				}
+				dataModel.put("input_list$" + inputName,
+						processedElements.entrySet().stream().sorted(Map.Entry.comparingByKey())
+								.map(Map.Entry::getValue).toArray(String[]::new));
+			}
+		}
+
+		// next we check for advanced input groups if they are defined, we process them and add to data model
+		if (!toolboxBlock.getRepeatingAdvancedInputs().isEmpty()) {
+			for (AdvancedInput advancedInput : toolboxBlock.getRepeatingAdvancedInputs()) {
+				Map<String, Element> matchingElements = elements.stream()
+						.filter(e -> e.getNodeName().equals("value") && e.getAttribute("name")
+								.startsWith(advancedInput.name()) && e.getAttribute("name")
+								.substring(advancedInput.name().length()).replaceAll("\\d", "").equals(""))
+						.collect(Collectors.toMap(e -> e.getAttribute("name"), e -> e));
+				Map<Integer, String> processedElements = new HashMap<>();
+				int idx = 0;
+				while (!matchingElements.isEmpty()) {
+					if (matchingElements.containsKey(advancedInput.name() + idx)) {
+						// check if nesting statement block that already provides any dependency with
+						// a same name, to avoid compile errors due to variable redefinitions
+						if (advancedInput.provides != null) {
+							for (Dependency dependency : advancedInput.provides) {
+								if (master.checkIfDepProviderInputsProvide(dependency)) {
+									master.addCompileNote(new BlocklyCompileNote(BlocklyCompileNote.Type.ERROR,
+											L10N.t("blockly.errors.duplicate_dependencies_provided",
+													advancedInput.name() + idx)));
+									return; // no need to do further processing, this needs to be resolved first by the user
+								}
+							}
+						}
+
+						AdvancedInput input = new AdvancedInput(advancedInput.name() + idx);

```

Comment:
Why is a redefinition of the AdvancedInput with index here needed? Not saying it is not, just asking why and suggesting to add a comment here so it is more clear

Reply:
Unlike normal inputs that don't provide dependencies but the code only, advanced/statement inputs are used to check for dependencies duplicates, so they have to be re-instantiated with a new name to form the block tree correctly.

Reply:
Well, that could be a reason to simplify this code.

Reply:
I think there is no need to make a copy of this object for the stack so you should try this :)

Reply:
✔️ Fixed